From c555fd74c009242c3864576bd5f17a1f8f4fdffd Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 18 Feb 2020 16:28:56 -0500
Subject: [PATCH] pesigcheck: Fix a wrong assignment

gcc says:

  pesigcheck.c: In function 'check_signature':
  pesigcheck.c:321:17: error: implicit conversion from 'enum <anonymous>' to 'enum <anonymous>' [-Werror=enum-conversion]
  321 |    reason->type = siBuffer;
      |                 ^
  pesigcheck.c:333:17: error: implicit conversion from 'enum <anonymous>' to 'enum <anonymous>' [-Werror=enum-conversion]
  333 |    reason->type = siBuffer;
      |                 ^
  cc1: all warnings being treated as errors

And indeed, that line of code makes no sense at all - it was supposed to
be reason->sig.type.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/pesigcheck.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pesigcheck.c b/src/pesigcheck.c
index 524cce307bf..8fa0f1ad03d 100644
--- a/src/pesigcheck.c
+++ b/src/pesigcheck.c
@@ -318,7 +318,7 @@ check_signature(pesigcheck_context *ctx, int *nreasons,
 			reason->type = SIGNATURE;
 			reason->sig.data = data;
 			reason->sig.len = datalen;
-			reason->type = siBuffer;
+			reason->sig.type = siBuffer;
 			nreason += 1;
 			is_invalid = true;
 		}
@@ -330,7 +330,7 @@ check_signature(pesigcheck_context *ctx, int *nreasons,
 			reason->type = SIGNATURE;
 			reason->sig.data = data;
 			reason->sig.len = datalen;
-			reason->type = siBuffer;
+			reason->sig.type = siBuffer;
 			nreason += 1;
 			has_valid_cert = true;
 		}
-- 
2.24.1

